geode-apis-compatible-with-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/sortedset/AbstractZRemRangeByLexIntegrationTest.java

```diff
@@ -0,0 +1,329 @@
+
+    int minLength = 5;
+    int maxLength = 8;
+    String min = StringUtils.repeat(BASE_MEMBER_NAME, minLength);
+    String max = StringUtils.repeat(BASE_MEMBER_NAME, maxLength);
+
+    List<String> sublist = new ArrayList<>(members.subList(minLength - 1, maxLength - 1));
+    members.removeAll(sublist);
+
+    // Range (v * 5) <= member name < (v * 8)
+    assertThat(jedis.zremrangeByLex(KEY, "[" + min, "(" + max)).isEqualTo(sublist.size());
+    assertThat(jedis.zrange(KEY, 0, -1)).hasSameElementsAs(members);
+  }
+
+  @Test
+  public void shouldReturnNumMembersRemoved_givenMultipleMembersInRange_withExclusiveMinAndInclusiveMax() {
+    List<String> members = populateSortedSet();
+
+    int minLength = 2;
+    int maxLength = 5;
+    String min = StringUtils.repeat(BASE_MEMBER_NAME, minLength);
+    String max = StringUtils.repeat(BASE_MEMBER_NAME, maxLength);
+
+    List<String> sublist = new ArrayList<>(members.subList(minLength, maxLength));
+    members.removeAll(sublist);
+
+    // Range (v * 2) < member name <= (v * 5)
+    assertThat(jedis.zremrangeByLex(KEY, "(" + min, "[" + max)).isEqualTo(sublist.size());
+    assertThat(jedis.zrange(KEY, 0, -1)).hasSameElementsAs(members);
+  }
+
+  @Test
+  public void shouldReturnNumMembersRemoved_givenMinusAndInclusiveMaxRange() {
+    List<String> members = populateSortedSet();
+
+    int maxLength = 8;
+    String max = StringUtils.repeat(BASE_MEMBER_NAME, maxLength);
+
+    List<String> sublist = new ArrayList<>(members.subList(0, maxLength));
+    members.removeAll(sublist);
+
+    // Range -infinity <= member name <= (v * 8)
+    assertThat(jedis.zremrangeByLex(KEY, "-", "[" + max)).isEqualTo(sublist.size());
+    assertThat(jedis.zrange(KEY, 0, -1)).hasSameElementsAs(members);
+  }
+
+  @Test
+  public void shouldReturnNumMembersRemoved_givenInclusiveMinAndPlusRange() {
+    List<String> members = populateSortedSet();
+
+    int minLength = 4;
+    String min = StringUtils.repeat(BASE_MEMBER_NAME, minLength);
+
+    List<String> sublist = new ArrayList<>(members.subList(minLength - 1, members.size()));
+    members.removeAll(sublist);
+
+    // Range (v * 4) <= member name < +infinity
+    assertThat(jedis.zremrangeByLex(KEY, "[" + min, "+")).isEqualTo(sublist.size());
+    assertThat(jedis.zrange(KEY, 0, -1)).hasSameElementsAs(members);
+  }
+
+  @Test
+  public void shouldReturnNumMembersRemoved_givenMinusAndPlusRange() {
+    List<String> members = populateSortedSet();
+
+    // Range -infinity <= member name < +infinity
+    assertThat(jedis.zremrangeByLex(KEY, "-", "+")).isEqualTo(members.size());
+    assertThat(jedis.zcard(KEY)).isEqualTo(0);
+  }
+
+  @Test
+  public void shouldRemoveMemberInRangeAndKey_givenOneMember() {
+    jedis.zadd(KEY, 1.0, "member");
+
+    assertThat(jedis.zremrangeByLex(KEY, "-", "+")).isEqualTo(1);
+    assertThat(jedis.exists(KEY)).isFalse();
+  }
+
+  @Test
+  public void shouldRemoveMembersInRange_givenMultipleMembers() {
```

Comment:
This behaviour is covered in the `shouldReturnNumMembersRemoved_givenMultipleMembersInRange_withInclusiveMinAndExclusiveMax()` test, so it shouldn't be necessary to repeat it here.